packages/camera/camera_android_camerax/lib/src/image_analysis.dart

```diff
@@ -0,0 +1,183 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:camera_platform_interface/camera_platform_interface.dart'
+    show CameraImageData, CameraImageFormat, CameraImagePlane, ImageFormatGroup;
+import 'package:flutter/services.dart' show BinaryMessenger;
+
+import 'android_camera_camerax_flutter_api_impls.dart';
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+import 'use_case.dart';
+
+/// Use case for providing CPU accessible images for performing image analysis.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/ImageAnalysis.
+class ImageAnalysis extends UseCase {
+  /// Creates an [ImageAnalysis].
+  ImageAnalysis(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      this.targetResolution})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = ImageAnalysisHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    _api.createFromInstance(this, targetResolution);
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+  }
+
+  /// Constructs an [ImageAnalysis] that is not automatically attached to a native object.
+  ImageAnalysis.detached(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      this.targetResolution})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = ImageAnalysisHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+  }
+
+  /// Stream that emits data whenever a frame is received for image streaming.
+  static final StreamController<CameraImageData>
```

Comment:
I see that you are using `StreamController`s for handling callback methods. I saw this in the  `onCameraClosing` PR as well. It looks like this will work in this scenario where you will only have one `ImageAnalysis` being used at a time, but it won't work for classes where callback methods would need to distinguish with instance is receiving the callback. Could you switch to passing a `Function` as the callback method? Similar to wrapper_example: https://github.com/bparrishMines/packages/blob/wrapper_example/packages/wrapper_example/lib/src/my_class.dart#L244

Reply:
@bparrishMines I re-wrapped the classes to use a callback!